core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -170,90 +168,107 @@ object DumpLogSegments {
   }
 
   /* print out the contents of the index */
-  private def dumpIndex(file: File,
-                        indexSanityOnly: Boolean,
-                        verifyOnly: Boolean,
-                        misMatchesForIndexFilesMap: mutable.HashMap[String, List[(Long, Long)]],
-                        maxMessageSize: Int) {
+  // Visible for testing
+  private[tools] def dumpIndex(file: File,
+                               indexSanityOnly: Boolean,
+                               verifyOnly: Boolean,
+                               misMatchesForIndexFilesMap: mutable.Map[String, List[(Long, Long)]],
+                               maxMessageSize: Int) {
     val startOffset = file.getName.split("\\.")(0).toLong
     val logFile = new File(file.getAbsoluteFile.getParent, file.getName.split("\\.")(0) + Log.LogFileSuffix)
     val fileRecords = FileRecords.open(logFile, false)
     val index = new OffsetIndex(file, baseOffset = startOffset, writable = false)
 
-    //Check that index passes sanityCheck, this is the check that determines if indexes will be rebuilt on startup or not.
-    if (indexSanityOnly) {
-      index.sanityCheck
-      println(s"$file passed sanity check.")
-      return
-    }
+    try {
 
-    for(i <- 0 until index.entries) {
-      val entry = index.entry(i)
-      val slice = fileRecords.slice(entry.position, maxMessageSize)
-      val firstRecord = slice.records.iterator.next()
-      if (firstRecord.offset != entry.offset + index.baseOffset) {
-        var misMatchesSeq = misMatchesForIndexFilesMap.getOrElse(file.getAbsolutePath, List[(Long, Long)]())
-        misMatchesSeq ::=(entry.offset + index.baseOffset, firstRecord.offset)
-        misMatchesForIndexFilesMap.put(file.getAbsolutePath, misMatchesSeq)
-      }
-      // since it is a sparse file, in the event of a crash there may be many zero entries, stop if we see one
-      if(entry.offset == 0 && i > 0)
+      //Check that index passes sanityCheck, this is the check that determines if indexes will be rebuilt on startup or not.
+      if (indexSanityOnly) {
+        index.sanityCheck()
+        println(s"$file passed sanity check.")
         return
-      if (!verifyOnly)
-        println("offset: %d position: %d".format(entry.offset + index.baseOffset, entry.position))
+      }
+
+      for (i <- 0 until index.entries) {
+        val entry = index.entry(i)
+
+        // since it is a sparse file, in the event of a crash there may be many zero entries, stop if we see one
+        if (entry.offset == index.baseOffset && i > 0)
+          return
```

Comment:
Nit: it will be good to have a log statement to print the file. Help user quickly identify such sparse files.

Reply:
This is pretty common and not a reason to be worried. If we want to highlight this, it's best to do it in a separate PR, I think.

Reply:
Are you OK with this @harshach?